### PR TITLE
feat: add Stripe to main join flow block

### PR DESCRIPTION
**Pull request by joaquimds**
## Description
- Adds Stripe components to front-end
- Adds Stripe token handling to back-end JoinService

## Motivation and Context
Part of the work to add Stripe support to the plugin.

## How Can It Be Tested?
1. Enable Stripe and add the 2 stripe keys on the settings page
2. Modify the payment plans and save them to trigger Stripe product creation
3. Sign up using the Join Form Block
4. Verify in the logs that a subscription was created

### CODE REVIEW

File: packages/join-flow/src/pages/payment-details.page.tsx [hunk 1]
```diff
@@ -242,3 +252,90 @@ const CreditCardPaymentPage: StagerComponent<FormSchema> = ({
...
+      handleError(submitError);
+      return;
+    }
+
+    const { error, confirmationToken } = await stripe.createConfirmationToken({
+      elements
+    });
+
+    // This point is only reached if there's an immediate error when
+    // creating the ConfirmationToken. Show the error to your customer (for example, payment details incomplete)
```
Comment (conatus) [inline packages/join-flow/src/pages/payment-details.page.tsx:323]: This reads like a copy and paste from Stripe's documentation (which it probably is!). Worth maybe re-phrasing it to fit more cleanly.

File: packages/join-flow/yarn.lock [hunk 2]
```diff
@@ -5674,6 +5674,11 @@ prepend-http@^2.0.0:
   resolved "https://registry.npmjs.org/prepend-http/-/prepend-http-2.0.0.tgz"
   integrity sha1-6SQ0v6XqjBn0HN/UAddBo8gZ2Jc=
 
+prettier@^3.3.3:
```
Comment (conatus) [inline packages/join-flow/yarn.lock:5677]: Do you want to add this to some kind of pre-commit?

**Review by conatus — CHANGES_REQUESTED** [review]
Thanks for this!

Some really minor questions and one change to a comment.

File: packages/join-flow/src/pages/payment-details.page.tsx [hunk 3]
```diff
@@ -242,3 +252,90 @@ const CreditCardPaymentPage: StagerComponent<FormSchema> = ({
...
+  data
+}) => {
+  const stripePromise = loadStripe(getEnv("STRIPE_PUBLISHABLE_KEY") as string);
+  const plan = (getEnv("MEMBERSHIP_PLANS") as any[]).find(
+    (plan) => plan.value === data.membership
+  );
+  const amount = plan.amount ? plan.amount * 100 : 100;
+  const currency = plan.currency.toLowerCase() || "gbp";
+  const paymentMethodTypes = ['card']
+  if (currency === 'gbp') {
```
[the remark this answers is not in the document]
Comment (conatus) [reply packages/join-flow/src/pages/payment-details.page.tsx:267]: Feels like this needs a comment, as its surprising to me.

File: packages/join-flow/src/pages/payment-details.page.tsx [hunk 4]
```diff
@@ -46,18 +64,14 @@ const DirectDebitPaymentPage: StagerComponent<FormSchema> = ({
...
-  const organisation = getEnv('ORGANISATION_NAME');
+  const organisation = getEnv("ORGANISATION_NAME");
 
   return (
     <form
       className="form-content"
       noValidate
       onSubmit={form.handleSubmit(onCompleted)}
     >
-      <div>
```
[the remark this answers is not in the document]
Comment (joaquimds) [reply packages/join-flow/src/pages/payment-details.page.tsx:57]: No, the summary has been moved to the outer component (line 49)

File: packages/join-flow/yarn.lock (hunk 2 above)
Comment (joaquimds) [reply packages/join-flow/yarn.lock:5677]: No it's just for my IDE to find it.

**Pull request merged**